compiler/src/dmd/cppmangle.d

```diff
@@ -137,18 +137,27 @@ private struct Context
      * Returns:
      *   The previous state of this `Context` object
      */
-    private Context push(lazy RootObject next) @safe
+    private Context push(RootObject delegate() nothrow @safe next) nothrow @safe
     {
         auto r = this.res;
         if (r !is null)
-            this.res = next;
+            this.res = next();
+        return Context(this.ti, this.fd, r);
+    }
+
+    // ditto
+    private Context push(RootObject delegate() nothrow @system next) nothrow @system
+    {
+        auto r = this.res;
+        if (r !is null)
+            this.res = next();
```

Comment:
This kind of change is one that can be broken out into a separate PR - [refactor] Replace all lazy parameters with `@safe` and `@system` delegates.